comm/src/main/scala/coop/rchain/comm/rp/HandleMessages.scala

```diff
@@ -74,29 +74,25 @@ object HandleMessages {
           m => handledWithMessage(ProtocolHelper.protocol(local).withPacket(m))
         )
 
+  def handleProtocolHandshakeResponse[F[_]: Monad: Metrics: ConnectionsCell: Log](
+      peer: PeerNode
+  ): F[CommunicationResponse] =
+    for {
+      _ <- Log[F].debug(s"Received protocol handshake response from $peer.")
+      _ <- ConnectionsCell[F].flatModify(_.addConn[F](peer))
+    } yield handledWithoutMessage
+
   def handleProtocolHandshake[F[_]: Monad: Time: TransportLayer: Log: ErrorHandler: ConnectionsCell: RPConfAsk: Metrics](
       peer: PeerNode,
-      protocolHandshake: ProtocolHandshake,
-      defaultTimeout: FiniteDuration
-  ): F[CommunicationResponse] = {
-
-    def notHandledHandshake(error: CommError): F[CommunicationResponse] =
-      Log[F]
-        .warn(s"Not adding. Could not receive response to heartbeat from $peer, reason: $error")
-        .as(notHandled(error))
-
-    def handledHandshake(local: PeerNode): F[CommunicationResponse] =
-      for {
-        _ <- ConnectionsCell[F].flatModify(_.addConn[F](peer))
-        _ <- Log[F].info(s"Responded to protocol handshake request from $peer")
-      } yield handledWithMessage(ProtocolHelper.protocolHandshakeResponse(local))
-
+      protocolHandshake: ProtocolHandshake
+  ): F[CommunicationResponse] =
     for {
-      local        <- RPConfAsk[F].reader(_.local)
-      hbrErr       <- TransportLayer[F].roundTrip(peer, ProtocolHelper.heartbeat(local), defaultTimeout)
-      commResponse <- hbrErr.fold(error => notHandledHandshake(error), _ => handledHandshake(local))
-    } yield commResponse
-  }
+      local    <- RPConfAsk[F].reader(_.local)
+      response = ProtocolHelper.protocolHandshakeResponse(local)
+      _        <- TransportLayer[F].send(peer, response) >>= ErrorHandler[F].fromEither
+      _        <- Log[F].info(s"Responded to protocol handshake request from $peer")
+      _        <- ConnectionsCell[F].flatModify(_.addConn[F](peer))
+    } yield handledWithoutMessage
```

Comment:
If it is async, then shouldn't the return be Unit? Is it because the heartbeat has to remain roundTrip?

Reply:
"handle **without ** message" - Once we remove `roundrripw` the `CommResponse` will be removed, and all those message will return unit